catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,26 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+        MetacardImpl metacard = getMetacard("dog_4leg_4paw_namespace.xml");
+        service.validate(metacard);
+        assertThat(service.getSchematronReport(), is(notNullValue()));
+        Pattern pattern = Pattern.compile("[\t \r\n]+");
+        for (String warning : service.getSchematronReport().getWarnings()) {
+            assertThat(pattern.matcher(warning).matches(), is(false));
```

Comment:
Well there are a couple ways to approach it, but think about what you're doing: You're replacing whitespace characters with a single space. How would you test that to make sure it works? Make sure the only whitespace characters that exist are spaces, and that there is only one of them.

Confusion about what you are doing is why I suggested testing the regex itself, which would be testing your "sanitize" method. Make a list of strings that have whitespace, newlines, tabs, and stuff like that, and actually assert things about what you expect. This way you will actually see in the code what you are trying to do.